haskellPackages.hindent: unbreak by fetching more recent source

markus1189 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change
Hindent fails to build due to a missing `MonadFail` instance, which is already present at HEAD, but not yet released.

There is an open issue for a release here: https://github.com/mihaimaruseac/hindent/issues/570

Meanwhile, we can fix `hindent` by fetching a more recent version.

Also, enable the tests, because they pass.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

peti committed:
ghc-9.0.x: add plenty of overrides to make packages build

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-21-g59d310e from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/2a9259d4b9119c690c17f5979d3a83f830c60608.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-22-g3b35a5f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/d64cb5e3a1e65146695772f9c92cc189dc7619e9.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-22-g3b35a5f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/80ba8be3dfe0bdb2014e232da9b8b14ba5630e02.

markus1189 committed:
microaeson: Unbreak via jailbreak

markus1189 committed:
haskellPackages.stan: Mark as not broken and enable tests

With https://github.com/NixOS/nixpkgs/pull/101008, this package builds
successfully and passes all tests as well.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-25-gc299817 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/1da79dfc6ce9663edddbf6a73811f760bd2ebf78.

cdepillabout committed:
Merge pull request #101008 from markus1189/haskell-unbreak-microaeson

microaeson: Unbreak via jailbreak

cdepillabout committed:
Merge pull request #101052 from markus1189/haskell-unbreak-stan

haskellPackages.stan: Mark as not broken and enable tests

markus1189 committed:
haskellPackages.hindent: unbreak by patch

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-25-gc299817 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/19db652d269877c06c18bba3a858b709b86bc714.

cdepillabout commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -630,14 +630,21 @@ self: super: {
   # Make elisp files available at a location where people expect it.
   hindent = (overrideCabal super.hindent (drv: {
+    # 2020-10-19: needs MonadFail fix, see https://github.com/mihaimaruseac/hindent/issues/570
+    src = pkgs.fetchFromGitHub {
+      owner = "mihaimaruseac";
+      repo = "hindent";
+      rev = "085417feebf6f24ec211a05eee6f28c3d7cc1caf";
+      sha256 = "0336k34549ni3faz43fp5bpbgx85v6v4fjcincylwggwylfnnill";
+    };
```
Normally we try to apply patches from upstream rather than overwriting `src`.  This so that as soon as it it fixed upstream, the derivation starts failing to build (because the patches no longer apply), so we immediately know that the patches can be removed.

If possible, can you rework this to apply whatever the patches from upstream that are needed for supporting later GHCs?

cdepillabout replied:
Oh, also, is https://github.com/mihaimaruseac/hindent the new home of `hindent`?  The latest hackage package (https://hackage.haskell.org/package/hindent-5.3.1) lists https://github.com/commercialhaskell/hindent as the home... 

which redirects to https://github.com/mihaimaruseac/hindent when you click on it...  Okay, ignore this question then!

markus1189 replied:
@cdepillabout I tried the patch way, but after picking more than 3 prs from GitHub which are all necessary to fix this I gave up because there are still more changes required to fix hindent.

The last upload on hackage was on 2019-06-28T08:35:20Z so there are quite some changes compared to `master` unfortunately...

cdepillabout replied:
As long as everything has been merged into `master`, you should be able to use a URL like the following:

https://github.com/mihaimaruseac/hindent/compare/5.3.1...75e6f1038dfbbe8ebd9481a5df51586b780a4e74.patch

This generates a patch file between two arbitrary commits/branches/tags.  Is it possible to use something like this to get all the commits you need?

markus1189 replied:
That's what I tried, but if fails due to changes to e.g. `travis.yml` files which are not present in the source fetched from hackage :sweat_smile:  Any ideas about that? :grin:

cdepillabout replied:
Hmm, I vaguely remember that you can pass flags to `fetchpatch` that causes it to ignore certain files.  Maybe that would work??

markus1189 replied:
Super tip, I tried that but for some reason the downloaded patch from GitHub for the correct range does not work due to a line being changed twice and leading to patch rejections.

I manually fixed the patch and added it to the patches folder.

Thanks for the hints!

markus1189 commented:
Yes makes sense, that's a better way. I will change this later today.

On Tue, Oct 20, 2020, 03:01 Dennis Gosnell <notifications@github.com> wrote:

> *@cdepillabout* commented on this pull request.
> ------------------------------
>
> In pkgs/development/haskell-modules/configuration-common.nix
> <https://github.com/NixOS/nixpkgs/pull/101082#discussion_r508147133>:
>
> > +    src = pkgs.fetchFromGitHub {
> +      owner = "mihaimaruseac";
> +      repo = "hindent";
> +      rev = "085417feebf6f24ec211a05eee6f28c3d7cc1caf";
> +      sha256 = "0336k34549ni3faz43fp5bpbgx85v6v4fjcincylwggwylfnnill";
> +    };
>
> Oh, also, is https://github.com/mihaimaruseac/hindent the new home of
> hindent? The latest hackage package (
> https://hackage.haskell.org/package/hindent-5.3.1) lists
> https://github.com/commercialhaskell/hindent as the home...
>
> which redirects to https://github.com/mihaimaruseac/hindent when you
> click on it... Okay, ignore this question then!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/101082#discussion_r508147133>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEQNT7I6PKZ2IV5AOTK6ITSLTOPHANCNFSM4SWPCFDA>
> .
>

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-25-gc299817 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/fcfa288f441d0f2c77f3874e6833931dba7e40a6.

peti commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -630,14 +630,16 @@ self: super: {
   # Make elisp files available at a location where people expect it.
   hindent = (overrideCabal super.hindent (drv: {
+    # 2020-10-19: needs MonadFail fix, see https://github.com/mihaimaruseac/hindent/issues/570
+    patches = [ ./patches/hindent-monadfail.patch ];
```
Please fetch that patch from Github instead of checking it into our repository.
